src/kernel/coinstats.cpp

```diff
@@ -69,32 +89,13 @@ DataStream TxOutSer(const COutPoint& outpoint, const Coin& coin)
 //! It is also possible, though very unlikely, that a change in this
 //! construction could cause a previously invalid (and potentially malicious)
 //! UTXO snapshot to be considered valid.
-static void ApplyHash(HashWriter& ss, const uint256& hash, const std::map<uint32_t, Coin>& outputs)
-{
-    for (auto it = outputs.begin(); it != outputs.end(); ++it) {
-        if (it == outputs.begin()) {
-            ss << hash;
-            ss << VARINT(it->second.nHeight * 2 + it->second.fCoinBase ? 1u : 0u);
-        }
-
-        ss << VARINT(it->first + 1);
-        ss << it->second.out.scriptPubKey;
-        ss << VARINT_MODE(it->second.out.nValue, VarIntMode::NONNEGATIVE_SIGNED);
-
-        if (it == std::prev(outputs.end())) {
-            ss << VARINT(0u);
```

Comment:
In commit "coinstats: Fix hash_serialized2 calculation" (351370a1d211615e3d5b158ccb0400cd79c5c085)

It seems reasonable to to get rid of `ss << VARINT(0)` added to the hash between transactions. It seems to date back to the original hashing code added in #7756 and I guess along with the nonzero `ss << VARINT(it->first + 1)` above was intended to delimit the boundary between transactions. None of this is necessary with the new hash format that repeats the transaction id before each coin.